woocommerce/woocommerce-ios
# [IAP] Update Site's networking layer with `isSiteOwner`

## Description
This PR updates the Site's networking layer to decode the `capabilities` container from the API response. We're specifically looking for the `own_site` property, in order to know if the current user is the site's owner.

## Testing instructions
1. Using proxyman or any other tool to intercept the network traffic, run the app and look for `https://public-api.wordpress.com/rest/v1.1/sites/$SITE_ID` (generally comes appended with `?http_envelope=1&force=wpcom`)
2. Inspect the request, and look for the capabilities container, will look like:
```
 {
      "ID": 219931069,
      "name": "20230608",
      "description": "",
      "URL": "https://woo-valiantly-insightful-otter.wpcomstaging.com",
      "capabilities": {
         ...
         ...
        "own_site": true,
         ...
         ...
```
3. Inside this container, the `own_site` property will be `true` when you're the site's owner, and `false` otherwise.
4. Unit Tests should pass
